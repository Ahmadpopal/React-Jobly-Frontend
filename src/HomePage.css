.HomePage-main-div {
  background-color: rgb(96, 165, 204);
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
  border: 1px solid black;
  border-radius: 10px;
  height: 30rem;
  margin: 5rem;
  text-align: center;
}

.HomePage-div-title {
  margin-top: 8rem;
}

.HomePage-login-button {
  margin: 10px;
  width: 7rem;
  color: white;
}

.HomePage-login-link {
  color: white;
}

.HomePage-login-link:hover {
  color: white;
  text-decoration: none;
}

.HomePage-signup-button {
  margin: 10px;
  color: white;
}

.HomePage-signup-link {
  color: white;
}

.HomePage-signup-link:hover {
  color: white;
  text-decoration: none;
}
