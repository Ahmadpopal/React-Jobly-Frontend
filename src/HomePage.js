import React, {useContext}  from 'react';
import {Link } from 'react-router-dom'

import {Container, Row, Col, Button } from 'reactstrap';

import UserContext from './Context'

import './HomePage.css'

const HomePage = () => {
    const { currentUser } = useContext(UserContext);

// HomePage:
//     - If user is Logged In Renders a Welcome back Container
//     - If user is Not not Logged In display a Page where Users can Log In or Sign Up for the Website 
 
    return (
        <div>
            {currentUser
            ? 
            <Container>
                <Row>
                    <Col className="HomePage-main-div">
                        <Row>
                            <Col className="HomePage-div-title">
                                <h3>Welcome back to Home Page {currentUser.username}</h3>
                            </Col>
                        </Row>
                        
                    </Col>
                </Row>
                
            </Container>
            :
            <Container>
                <Row>
                    <Col className="HomePage-main-div">
                        <Row>
                            <Col className="HomePage-div-title">
                                <h3>Welcome to Jobly Home Page.</h3>
                                <h3>Open jobs.</h3>
                                <h3>People hiring.</h3>
                            </Col>
                        </Row>
                        <Row>
                            <Col >
                                <Button className="HomePage-login-button" color="success">
                                    <Link className="HomePage-login-link"  to='/login'>
                                        Log In
                                    </Link>
                                </Button >
                                <Button className="HomePage-signup-button" color="primary">
                                    <Link className="HomePage-signup-button"  to='/register'>
                                        Sign Up
                                    </Link>
                                </Button>
                            </Col>
                        </Row>
                    </Col>
                </Row>
            </Container>            
        
        }
            {/* <Container>
                <Row>
                    <Col className="HomePage-main-div">
                        <Row>
                            <Col className="HomePage-div-title">
                                <h3>Welcome to Jobly Home Page.</h3>
                                <h3>Open jobs.</h3>
                                <h3>People hiring.</h3>
                            </Col>
                        </Row>
                        <Row>
                            <Col >
                                <Button className="HomePage-login-button" color="success">
                                    <Link className="HomePage-login-link"  to='/login'>
                                        Log In
                                    </Link>
                                </Button >
                                <Button className="HomePage-signup-button" color="primary">
                                    <Link className="HomePage-signup-button"  to='/register'>
                                        Sign Up
                                    </Link>
                                </Button>
                            </Col>
                        </Row>
                    </Col>
                </Row>
            </Container>             */}
        </div>
    )
}


export default HomePage;