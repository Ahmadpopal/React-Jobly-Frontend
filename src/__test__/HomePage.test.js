import React, { useContext } from 'react';
import {render, fireEvent, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import HomePage from '../HomePage'
import UserContext from '../Context';


// SMOKE TEST
    // - we create currentUser and set it to null 
    // - check if Home Page renders without Crushing 
test('HomePage Component should render without crushing', ()=> {
    let currentUser = null;

	render(
        <UserContext.Provider value={{ currentUser }}>
            <MemoryRouter>
                <HomePage />
            </MemoryRouter>
        </UserContext.Provider>
    );
});


// SNAPSHOT TEST 

// IF USER IS NOT LOG IN 
it('matches snapshot', () => {
    let currentUser = null;

    const { asFragment } = render(
        <UserContext.Provider value={{ currentUser }}>
            <MemoryRouter>
                <HomePage />
            </MemoryRouter>
        </UserContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});

