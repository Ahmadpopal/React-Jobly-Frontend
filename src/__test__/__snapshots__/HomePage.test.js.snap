// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="HomePage-main-div col"
        >
          <div
            class="row"
          >
            <div
              class="HomePage-div-title col"
            >
              <h3>
                Welcome to Jobly Home Page.
              </h3>
              <h3>
                Open jobs.
              </h3>
              <h3>
                People hiring.
              </h3>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col"
            >
              <button
                class="HomePage-login-button btn btn-success"
              >
                <a
                  class="HomePage-login-link"
                  href="/login"
                >
                  Log In
                </a>
              </button>
              <button
                class="HomePage-signup-button btn btn-primary"
              >
                <a
                  class="HomePage-signup-button"
                  href="/register"
                >
                  Sign Up
                </a>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
